File: Nogle/Bulls_and_Cows.java
import java.util.*;

public class Bulls_and_Cows {
	
	public void solution(String S, String G) {
		
		int A = 0;
		int B = 0;
		
		ArrayList<Integer> secret = new ArrayList<Integer>();
		ArrayList<Integer> guess = new ArrayList<Integer>();
		
		for(int i = 0; i < S.length(); i++) {
			secret.add(S.charAt(i) - '0');
			guess.add(G.charAt(i) - '0');
		}
		
		for(int i = 0; i < secret.size(); i++) {
			if(secret.get(i) == guess.get(i)) {
				A++;
				secret.remove(i);
				guess.remove(i);
				i--;
			}
		}
		
		for(int i = 0; i < guess.size(); i++) {
			for(int j = 0; j < secret.size(); j++) {
				if(guess.get(i) == secret.get(j)) {
					secret.remove(j);
					B++;
					break;
				}
			}
		}
		
		System.out.print(A + "A" + B + "B" + "\n");
		
	}
	
	public static void main(String[] args) {
		
		String secret = "7807";
		String guess = "7810";
		Bulls_and_Cows BC = new Bulls_and_Cows();
		BC.solution(secret, guess);
	}
}

File: Nogle/Bulls_and_Cows.py
import os

def solution(S, G):
	A = 0
	B = 0
	A_check = []
	secret = list(S)
	guess = list(G)
	
	for i in range(0,len(secret)):	
		if secret[i] == guess[i]:
			A = A + 1
			A_check.append(i)
	
	#print(A_check)
	count = 0
	for i in range(0, len(A_check)):
		del secret[A_check[i]-count]
		del guess[A_check[i]-count]
		count += 1
		print(secret)
		print(guess)

	
	for i in range(0, len(guess)):
		for j in range(0, len(secret)):
			if guess[i] == secret[j]:
				del secret[j]
				B = B + 1
				break
				
	print(str(A) + "A" + str(B) + "B")	
	
def main():
	secret = "7807"
	guess = "7810"
	solution(secret, guess)

if __name__ == '__main__':
	main()